mantidproject/mantid
# Fix incorrect BayesQuasi calculation

### Description of work
This PR fixes a bug introduced in a recent PR (#37975), where the result workspace was not being constructed correctly after a BayesQuasi calculation using the quasielasticbayes package.

I have added a system test to so we have better test coverage for this algorithm.

*There is no associated issue.*

**Report to:** Spencer Howells

### To test:
1. Open Inelastic Bayes Fitting
2. Go to Quasi tab
3. Load following sample and resolution

[irs26176_graphite002_red.zip](https://github.com/user-attachments/files/17060962/irs26176_graphite002_red.zip)

[irs26173_graphite002_res.zip](https://github.com/user-attachments/files/17060965/irs26173_graphite002_res.zip)

6. Click `Run`
7. Plot all in the Output options section of the GUI
8. You should get plots similiar to these two:
![image](https://github.com/user-attachments/assets/1eab9abe-92a0-42f1-ab68-8e5093920e71)


*This does not require release notes* because **this is a regression**


### Reviewer

Please comment on the points listed below ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)).
**Your comments will be used as part of the gatekeeper process, so please comment clearly on what you have checked during your review.** If changes are made to the PR during the review process then your final comment will be the most important for gatekeepers. In this comment you should make it clear why any earlier review is still valid, or confirm that all requested changes have been addressed.

#### Code Review

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Do the release notes conform to the [release notes guide](https://developer.mantidproject.org/Standards/ReleaseNotesGuide.html)?

#### Functional Tests

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### Gatekeeper

If you need to request changes to a PR then please add a comment and set the review status to "Request changes". This will stop the PR from showing up in the list for other gatekeepers.
